finesse/hardware/plugins/spectrometer/ftsw500_interface.py

```diff
@@ -0,0 +1,194 @@
+"""Module containing code for sending commands to FTSW500 for the ABB spectrometer.
+
+Communication is via TCP.
+
+The FTSW500 program must be running on the computer at FTSW500_HOST for the commands to
+work.
+"""
+
+import logging
+from socket import AF_INET, SOCK_STREAM, socket
+
+from PySide6.QtCore import QTimer
+
+from finesse.config import (
+    FTSW500_HOST,
+    FTSW500_POLLING_INTERVAL,
+    FTSW500_PORT,
+    FTSW500_TIMEOUT,
+)
+from finesse.hardware.plugins.spectrometer.ftsw500_interface_base import (
+    FTSW500Error,
+    FTSW500InterfaceBase,
+)
+from finesse.spectrometer_status import SpectrometerStatus
+
+
+def parse_response(response: bytes) -> SpectrometerStatus | None:
+    r"""Parse FTSW500's response.
+
+    The server always returns a response after having received a command. The response
+    will start with the tag "ACK" in case of success, or "NAK" otherwise, and is
+    terminated by the end-of-line character "\n". An answer can also contain a string
+    value or message after the tag, which will be preceded by "&". For example, querying
+    whether the instrument is currently measuring data would yield a response
+    "ACK&true\n" or "ACK&false\n".
+
+    Querying the FTSW500 state yields one of the following values:
+    0: when disconnected
+    1: when in the process of connecting to an instrument
+    2: when acquiring data without saving it
+    3: when acquiring and saving data
+    -1: when in an intermediate state that should normally not last for a long time
+        (less than 500 ms) or when the FTSW500_SDK object is not well initialized
+
+    Args:
+        response: the byte sequence received from FTSW500
+
+    Returns:
+        SpectrometerStatus: the generic spectrometer device state of FTSW500
+    """
+    msg = response.decode()
+    if "NAK" in msg:
```

Comment:
I think (personally) the following makes it clearer what a message looks like. Feel free to ignore -- kinda subjective.
```suggestion
    if msg[:3] == "NAK"
```

Reply:
Yeah, good idea

Reply:
What about:
```suggestion
    if msg.startswith("NAK"):
```

Reply:
There really is a Python method for everything!